ql/pricingengines/bond/bondfunctions.hpp

```diff
@@ -118,7 +118,7 @@ namespace QuantLib {
                             const YieldTermStructure& discountCurve,
                             Date settlementDate = Date(),
                             Real price = Null<Real>(),
-                            const Bond::Price::Type priceType = Bond::Price::Clean);
+                            Bond::Price::Type priceType = Bond::Price::Clean);
```

Comment:
@lballabio This is an automated PR to my repo by clang-tidy. If you have spare cycles, please explain to me why it's recommending that the `const` keyword be dropped.

Reply:
https://clang.llvm.org/extra/clang-tidy/checks/readability-avoid-const-params-in-decls.html

There's not a lot of content at that link, but as far as I can see the reason is that it has no effects on the interface so it can be omitted to increase the signal/noise ratio.

Reply:
Thank you. I think I get it. `Bond::Price::Type` is a value type, not a pointer, so declaring it as `const` has no effect. `const` affects only pointers (non-top level) which modifies them so that the object being pointed to cannot be changed. Do you agree with that statement?

Reply:
I agree.

Reply:
Thanks, Meistro!